File: src/models/agent-provider-model.ts
/** Types cho quản lý AI Agent Provider Model — khớp với DTO ở
 * `src-tauri/src/models/agent_provider_model.rs`. */

export interface AgentProviderModel {
  id: number;
  provider_id: number;
  /** Tên provider (join sẵn để hiển thị). */
  provider_name: string;
  name: string;
  code: string;
  version: string;
  description: string;
  /** Có cho phép sử dụng model trong hệ thống hay không. */
  enabled: boolean;
  created_at: string;
  updated_at: string;
}

/** Request đăng ký / cập nhật model. */
export interface AgentProviderModelRequest {
  provider_id: number;
  name: string;
  code: string;
  version: string;
  description: string;
  enabled: boolean;
}

/** Nhãn hiển thị cho 1 model — dùng thẳng tên (đã đầy đủ, vd "Claude Opus 4.8"). */
export function agentProviderModelLabel(model: AgentProviderModel): string {
  return model.name;
}

/** Cờ `--model` truyền cho CLI — ưu tiên mã model (`code`), fallback về tên. */
export function agentProviderModelFlag(model: AgentProviderModel): string {
  return model.code || model.name;
}

File: src-tauri/src/models/agent_provider_model.rs
//! Model cho quản lý AI Agent Provider Model — khai báo các model của từng
//! provider được phép sử dụng trong hệ thống (VD: Claude → opus/sonnet/haiku).
//! Lưu ở bảng `agent_provider_models` (PostgreSQL), truy cập qua stored procedure.

use serde::{Deserialize, Serialize};

fn default_string() -> String {
    String::new()
}

fn default_true() -> bool {
    true
}

/// Một model thuộc về một AI Agent Provider.
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct AgentProviderModel {
    pub id: i32,
    pub provider_id: i32,
    /// Tên provider (join sẵn để hiển thị trên danh sách).
    #[serde(default = "default_string")]
    pub provider_name: String,
    pub name: String,
    #[serde(default = "default_string")]
    pub code: String,
    #[serde(default = "default_string")]
    pub version: String,
    #[serde(default)]
    pub description: String,
    #[serde(default = "default_true")]
    pub enabled: bool,
    pub created_at: String,
    pub updated_at: String,
}

/// Request đăng ký / cập nhật model.
#[derive(Debug, Deserialize)]
pub struct AgentProviderModelRequest {
    pub provider_id: i32,
    pub name: String,
    #[serde(default = "default_string")]
    pub code: String,
    #[serde(default = "default_string")]
    pub version: String,
    #[serde(default)]
    pub description: String,
    #[serde(default = "default_true")]
    pub enabled: bool,
}
